### ISSUE TITLE
Scrap the caption of a music video as lyrics 

### ISSUE THREAD

**Issue by smarttommyau** [z-huang/InnerTune#506]
### Checklist

- [X] I've checked that there is no other issue about this feature request.
- [X] This issue contains only one feature request.
- [X] The title of this issue accurately describes the feature request.

### Feature description

Scrap the caption of a music video as lyrics.
Should also be able to choose the language of the caption.

### Why do you want this feature?

As most music video are most likely not able to match lyrics(bc of their song and artist name most likely to be not accurate), so using the caption of it can make lyrics works for them, also using caption are most likely to synchronise to the video more accurate.

### Additional information

This open source project may be useful https://github.com/jdepoix/youtube-transcript-api, although it is wriiten in python, but it seems to be the most reputable source.

**Comment by eightyy8** [comment]
Smart idea!

**Comment by z-huang** [comment]
Not planned because it's too "[advanced](https://github.com/z-huang/InnerTune/discussions/421#discussioncomment-6198050)".

**Comment by smarttommyau** [comment]
This actually not advanced at all. It is quite minimal. Just need some parsing of the api. I just find out that an api call like that can actually get the caption easily, you make take a look. It's the same api InnerTune is using [https://music.youtube.com/youtubei/v1/get_transcript]()
Example code on using curl
` curl -s 'https://music.youtube.com/youtubei/v1/get_transcript?key=AIzaSyC9XL3ZjWddXya6X74dJoCTL-WEYFDNX3' -X POST -H 'Content-Type: application/json' --data-raw "{\"context\":{\"client\":{\"clientName\":\"WEB\",\"clientVersion\":\"2.2021111\"}},\"params\":\"$(printf '\n\x0b<VideoId>' | base64)\"}"`
Taking reference from stackoverfloiw [https://stackoverflow.com/a/70013529/19362495]()
I hope this can change your mind of putiing this as not planned.

**Comment by smarttommyau** [comment]
Yeah, but having something that is better than nothing when many of lyrics is not avaliable or not having the timestamp
